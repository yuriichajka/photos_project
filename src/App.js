import React, { useEffect, useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { BrowserRouter as Router, Link, Route, Switch } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { GlobalStyles } from './GlobalStyles';
import Modal from './components/Modal';
import Image from './components/Image';

import { getPhotos } from './services/api';

const App = () => {
  const { photos } = useSelector(state => state);
  const dispatch = useDispatch();
  let [showModal, setShowModal] = useState(false);

  useEffect(() => {
    getPhotos().then(value => dispatch({type: 'SET_PHOTOS', payload: value.data}))
  }, [dispatch])

    const openModal = () => {
        setShowModal(prev => !prev)
    }

  return (
    <>
        <Router>
            <Switch>
                <Route exact path={'/photo/:id'} render={() => <Modal showModal={showModal} setShowModal={setShowModal}/>}/>
            </Switch>
            <Container>
                <Row className="justify-content-md-center">
                    <Col lg={9} className='center'>
                        {
                            photos.map(value =>
                                <Link to={'/photo/' + value.id}>
                                    <Image item={value.url} key={value.id} onClick={openModal}/>
                                </Link>)
                        }
                    </Col>
                </Row>
            </Container>
            <GlobalStyles/>
        </Router>
    </>
  );
}
export default App;
