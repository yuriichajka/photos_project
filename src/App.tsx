import React, { useEffect, useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { BrowserRouter as Router, Link, Route, Switch } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { GlobalStyles } from './GlobalStyles';
import { Modal, Image } from './components'

import { getImages } from './redux/ducks/images';
import { RootState } from './redux/configureStore';

export interface IContext  {
    showModal: boolean
    toggleModal: () => void
}
// @ts-ignore
export const ModalContext = React.createContext<IContext>()

const App: React.FC = () => {
  const dispatch = useDispatch();

  let [showModal, setShowModal] = useState(false);

  const images = useSelector((state: RootState) => state.images.images);

  const toggleModal = () => {
      setShowModal((previousState) => !previousState);
  }

  useEffect(() => {
      dispatch(getImages())
  }, [dispatch]);

    return (
    <ModalContext.Provider value={{ showModal, toggleModal }}>
        <Router>
            <Switch>
                <Route
                    exact
                    path={'/photo/:id'}
                    render={() => (
                        <Modal />
                    )}
                />
            </Switch>
            <Container>
                <Row className="justify-content-md-center">
                    <Col lg={9} className='center'>
                        {images.map(value => (
                                <Link to={'/photo/' + value.id}>
                                    <Image item={value.url} key={value.id} onClick={toggleModal} />
                                </Link>
                        ))}
                    </Col>
                </Row>
            </Container>
            <GlobalStyles/>
        </Router>
    </ModalContext.Provider>
  );
}

export default App;
