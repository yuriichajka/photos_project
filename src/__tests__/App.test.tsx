import React from 'react';
import App from '../App';
import { Provider } from 'react-redux'
import store from '../redux/configureStore';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('snapp',  () => {
    let wrapper;
    it('snapshot',  () => {
        wrapper = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    describe('dawd', () => {
        beforeEach(() => {
            wrapper = mount(<Provider store={store}><App /></Provider>);
        })

        it('should render something',  () => {
            expect(wrapper).not.toBeNull();
        });

        it('should be container', () => {
            expect(wrapper.find('.container')).toHaveLength(1);
        })
    })
});
