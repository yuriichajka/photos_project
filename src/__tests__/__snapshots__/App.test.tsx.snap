// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should add a snapshot 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="center col-lg-9"
    />
  </div>
</div>
`;
